### PR TITLE
Internal Solution for Guess the Animal

### PR DESCRIPTION
**Pull request by brad182**
Copy pasted Brian Dean's code, added my own Java implementation, changed the .json file

### CODE REVIEW

File: solutions/usaco-893.mdx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	vector<string> &v1 = characteristics[i], &v2 = characteristics[j];
+	for (int i=0; i<v1.size(); i++)
+	for (int j=0; j<v2.size(); j++)
+		if (v1[i] == v2[j]) count++;
+	return count;
+}
+ 
+int main(void)
+{
+	ifstream fin ("guess.in");
```
Comment (nwatx) [inline solutions/usaco-893.mdx:37]: would be nice to have this w/ the guide template

File: solutions/usaco-893.mdx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	fin >> N;
+	string s;
+	for (int i=0; i<N; i++) {
+	int K;
```
Comment (maggie-j-liu) [inline solutions/usaco-893.mdx:41]: indentation seems a little messed up

File: solutions/usaco-893.mdx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+	fin >> s >> K;
+	for (int j=0; j<K; j++) {
```
Comment (maggie-j-liu) [inline solutions/usaco-893.mdx:43]: could you edit this code and add spaces around all the operators? like
```suggestion
	for (int j = 0; j < K; j++) {
```

File: solutions/usaco-893.mdx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+author: Brad Ma
+---
+
+[Official Analysis (C++)](https://codeforces.com/blog/entry/22712)
+
+<LanguageSection>
+  
+<CPPSection>
+  
+```cpp
```
Comment (maggie-j-liu) [inline solutions/usaco-893.mdx:14]: would be nice to add comments to this code instead of just copy pasting

same for java sol

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

File: solutions/usaco-893.mdx (hunk 2 above)
Comment (dongliu0426) [reply solutions/usaco-893.mdx:41]: ^ not addressed

File: solutions/usaco-893.mdx (hunk 3 above)
Comment (dongliu0426) [reply solutions/usaco-893.mdx]: ^ not addressed

File: solutions/usaco-893.mdx [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+		}
+	}
+    
+	int largest = 0;
+	for (int i = 0; i < N; i++)
+		for (int j=i+1; j<N; j++)
+			largest = max(largest, num_common(i,j)); // keep track of the largest
+ 
+	ofstream fout ("guess.out");
+	fout << largest+1 << "\n"; // output one more than the largest
```
Comment (maggie-j-liu) [inline solutions/usaco-893.mdx:55]:
```suggestion
		for (int j = i + 1; j < N; j++)
			largest = max(largest, num_common(i, j)); // keep track of the largest
 
	ofstream fout ("guess.out");
	fout << largest + 1 << "\n"; // output one more than the largest
```

Comment (brad182) [reply solutions/usaco-893.mdx:55]: oops i missed that one

File: solutions/usaco-893.mdx [hunk 6]
```diff
@@ -0,0 +1,123 @@
+---
+id: usaco-893
+source: USACO Bronze 2019 January
+title: Guess the Animal
+author: Brad Ma
+---
+
+[Official Analysis (C++)](https://codeforces.com/blog/entry/22712)
```
Comment (dongliu0426) [inline solutions/usaco-893.mdx:8]: link to usaco one (?)

Comment (maggie-j-liu) [inline solutions/usaco-893.mdx:8]: wrong copy paste?

Comment (brad182) [reply solutions/usaco-893.mdx:8]: oh yeah good catch i copy pasted a template from a codeforces problem and forgot to change

**Review by dongliu0426 — APPROVED** [review]

File: solutions/usaco-893.mdx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+```cpp
+#include <iostream>
+#include <fstream>
+#include <vector>
+using namespace std;
+ 
+int N;
+vector<string> characteristics[100];
+
+// Official Analysis Code by Brian Dean
```
Comment (maggie-j-liu) [inline solutions/usaco-893.mdx:23]: maybe move this to the top? seems kind of out of place here

**Review by HarryWangATX — CHANGES_REQUESTED** [review]
wheres the implementation header??

File: solutions/usaco-893.mdx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+source: USACO Bronze 2019 January
+title: Guess the Animal
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://usaco.org/current/data/sol_guess_bronze_jan19.html)
+
+<LanguageSection>
+	
+## Implementation
```
Comment (HarryWangATX) [inline solutions/usaco-893.mdx:12]: outside of this

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/usaco-893.mdx [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+import java.util.*;
+import java.io.*;
+
+public class GuessTheAnimal {
+	static ArrayList<String>[] characteristics;
+
+	static int numCommon (int i, int j) {
+		int count = 0;
+		ArrayList<String> one = characteristics[i];
+		ArrayList<String> two = characteristics[j];
```
Comment (HarryWangATX) [inline solutions/usaco-893.mdx:77]: its good practice to use 'List' instead of 'ArrayList'

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

File: solutions/usaco-893.mdx [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+		for (int x = 0; x < characteristics.length; x++) { // initialize lists
+			characteristics[x] = new ArrayList<>();
+		}
+
+		for (int i = 0; i < n; i++) { // read input
+			String s = io.next();
+			int k = io.nextInt();
+
+			for (int x = 0; x < k; x++) {
+				String string = io.next();
```
Comment (HarryWangATX) [inline solutions/usaco-893.mdx:104]: thonk String string

File: solutions/usaco-893.mdx [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+				characteristics[i].add(string); // add to the array of lists
+			}
+		}
+
+		int largest = 0;
+		for (int i = 0; i < n; i++) { // loop through the combinations
+			for (int j = i + 1; j < n; j++) {
+				largest = Integer.max(largest, numCommon(i, j)); // keep track of the largest
```
Comment (HarryWangATX) [inline solutions/usaco-893.mdx:112]: should use Math.max

File: solutions/usaco-893.mdx [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+	static List<String>[] characteristics;
+
+	static int numCommon (int i, int j) {
+		int count = 0;
+		List<String> one = characteristics[i];
+		List<String> two = characteristics[j];
+
+		for (int x = 0; x < one.size(); x++) {
+			for (int y = 0; y < two.size(); y++) {
+				if (one.get(x).equals(two.get(y))) { // // if the same characteristic, add to count
```
Comment (HarryWangATX) [inline solutions/usaco-893.mdx:81]: why 2 //?

**Review by HarryWangATX — CHANGES_REQUESTED** [review]

**Review by HarryWangATX — APPROVED** [review]

**Pull request merged**